src/QsCompiler/CompilationManager/EditorSupport/CodeActions.cs

```diff
@@ -408,17 +408,27 @@ internal static IEnumerable<(string, WorkspaceEdit)> DocCommentSuggestions(this
         {
             var overlapping = file?.FragmentsOverlappingWithRange(range);
             var fragment = overlapping?.FirstOrDefault();
-            if (fragment?.Kind == null || overlapping.Count() > 1) return Enumerable.Empty<(string, WorkspaceEdit)>(); // only suggest doc comment directly on the declaration
+            if (fragment?.Kind == null || overlapping.Count() != 1) return Enumerable.Empty<(string, WorkspaceEdit)>(); // only suggest doc comment directly on the declaration
 
             var (nsDecl, callableDecl, typeDecl) = (fragment.Kind.DeclaredNamespace(), fragment.Kind.DeclaredCallable(), fragment.Kind.DeclaredType());
             var declSymbol = nsDecl.IsValue ? nsDecl.Item.Item1.Symbol 
                 : callableDecl.IsValue ? callableDecl.Item.Item1.Symbol
                 : typeDecl.IsValue ? typeDecl.Item.Item1.Symbol : null;
-            var declRange = fragment?.GetRange();
-            if (declSymbol == null || file.DocumentingComments(declRange.Start).Any()) return Enumerable.Empty<(string, WorkspaceEdit)>();
+            var declStart = fragment.GetRange().Start;
+            if (declSymbol == null || file.DocumentingComments(declStart).Any()) return Enumerable.Empty<(string, WorkspaceEdit)>();
+
+            // set declStart to the position of the first attribute attached to the declaration
+            bool EmptyOrFirstAttribute(IEnumerable<CodeFragment> line, out CodeFragment att)
+            {
+                att = line.Reverse().TakeWhile(t => t.Kind is QsFragmentKind.DeclarationAttribute).LastOrDefault();
+                return att != null || !line.Any();
+            }
+            var preceding = file.GetTokenizedLine(declStart.Line).TakeWhile(ContextBuilder.TokensUpTo(new Position(0, declStart.Character)));
+            for (var lineNr = declStart.Line; lineNr-- > 0 && EmptyOrFirstAttribute(preceding, out var precedingAttribute); preceding = file.GetTokenizedLine(lineNr))
+            { if (precedingAttribute != null) declStart = precedingAttribute.GetRange().Start.WithUpdatedLineNumber(lineNr+1); }
```

Comment:
This for loop is too dense. For readability, it should be reworked. The post- decrement operator should not be used except as a statement by itself. If possible, stick to the more standard use of the for loop:
for( init loop index; condition; loop index increment [or decrement] ) { ... }

Reply:
The braces should be on their own lines.